libs/features/icons/icons.js

```diff
@@ -49,26 +52,56 @@ function decorateToolTip(icon) {
   wrapper.parentElement.replaceChild(icon, wrapper);
 }
 
-export default async function loadIcons(icons, config) {
-  const iconSVGs = await fetchIcons(config);
-  if (!iconSVGs) return;
+export async function setNodeIndexClass(icons) {
   icons.forEach(async (icon) => {
-    const { classList } = icon;
-    if (classList.contains('icon-tooltip')) decorateToolTip(icon);
-    const iconName = icon.classList[1].replace('icon-', '');
-    const existingIcon = icon.querySelector('svg');
-    if (!iconSVGs[iconName] || existingIcon) return;
+    const parent = icon.parentNode;
+    const children = parent.childNodes;
+    const nodeIndex = Array.prototype.indexOf.call(children, icon);
+    let indexClass = (nodeIndex === children.length - 1) ? 'last' : 'middle';
+    if (nodeIndex === 0) indexClass = 'first';
+    if (children.length === 1) indexClass = 'only';
+    icon.classList.add(`node-index-${indexClass}`);
+  });
+}
+
+export default async function loadIcons(icons) {
+  const fedRoot = getFederatedContentRoot();
+  const iconRequests = [];
+  const iconsToFetch = new Map();
+
+  icons.forEach((icon) => {
+    icon.classList.add('milo-icon');
+    const iconName = [...icon.classList].find((c) => c.startsWith('icon-'))?.substring(5);
+    if (icon.dataset.svgInjected || !iconName) return;
+    if (iconName === 'tooltip') {
+      decorateToolTip(icon);
+      return;
+    }
+    if (!federalIcons[iconName] && !iconsToFetch.has(iconName)) {
+      const url = `${fedRoot}/federal/assets/icons/svgs/${iconName}.svg`;
```

Comment:
Does this mean, we switch from having one general purpose file
`const iconSVGs = await fetchIcons(config);` - to one icon per file now?

Reply:
@mokimo - That is correct. This way we have full author ability to add more icons w out a code change. I have this explained in the description, but please let me know if this is not clear.

Reply:
Nono that's fine, I'm looking at this from a performance lense... So far I've seen that requests add overhead and usually lead to degraded performance. We now potentially do *a lot* of requests.

<img width="990" alt="Screenshot 2024-10-02 at 11 09 30" src="https://github.com/user-attachments/assets/3eb45968-1c99-421f-b250-9fd89fb09b38">

I'm a little concerned about our mobile users, maybe we need to go back to reserving space and loading icons async? That's something you've been working on previously.